gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/OracleExtractor.java

```diff
@@ -0,0 +1,402 @@
+    
+    if ((limitStartIndex > 0 && !multiPredicate) || (limitStartIndex > 0 && multiPredicate && !leadingLimit)) {
+      outputQuery = query.substring(0, limitStartIndex) + " " + query.substring(limitEndIndex);
+    } else if (limitStartIndex > 0 && multiPredicate && leadingLimit) {
+      outputQuery = query.substring(0, limitStartIndex + 6) + " " + query.substring(limitEndIndex + 4);
+    }
+    return outputQuery.trim().replaceAll(" +", " ");
+  }
+
+  private static int getLimitEndIndex(String inputQuery, int limitStartIndex) {
+    int limitEndIndex = -1;
+    if (limitStartIndex > 0) {
+      limitEndIndex = inputQuery.charAt(limitStartIndex + 1) == 'w' ? limitStartIndex + 17 : limitStartIndex + 15;
+      String remainingQuery = inputQuery.substring(limitEndIndex);
+      boolean numFound = false;
+
+      int pos = 0;
+      while (pos < remainingQuery.length()) {
+        char ch = remainingQuery.charAt(pos);
+        if (ch == ' ' && !numFound) {
+          pos++;
+          continue;
+        } else if (numFound && (!Character.isDigit(ch))) {
+          break;
+        } else {
+          numFound = true;
+        }
+        pos++;
+      }
+      limitEndIndex = limitEndIndex + pos;
+    }
+    return limitEndIndex;
+  }
+
+  // @Override
+  // public String constructSampleClause() {
+  //   long sampleRowCount = this.getSampleRecordCount();
+  //   if (sampleRowCount >= 0) {
+  //     return " top " + sampleRowCount;
+  //   }
+  //   return "";
+  // }
+
+    @Override
+  public String constructSampleClause() {
+    long sampleRowCount = this.getSampleRecordCount();
+    if (sampleRowCount >= 0) {
+      return " rownum <= " + sampleRowCount;
+    }
+    return "";
+  }
+
+  @Override
+  public String getWatermarkSourceFormat(WatermarkType watermarkType) {
+    String columnFormat = null;
+    switch (watermarkType) {
+      case TIMESTAMP:
+        columnFormat = "yyyy-MM-dd HH:mm:ss";
+        break;
+      case DATE:
+        columnFormat = "yyyy-MM-dd";
+        break;
+      case SIMPLE:
+        break;
+      default:
+        log.error("Watermark type " + watermarkType.toString() + " not recognized");
+    }
+    return columnFormat;
+  }
+
+  @Override
+  public String getHourPredicateCondition(String column, long value, String valueFormat, String operator) {
+    log.debug("Getting hour predicate for Oracle");
+    String formattedvalue = Utils.toDateTimeFormat(Long.toString(value), valueFormat, HOUR_FORMAT);
+    return Utils.getCoalesceColumnNames(column) + " " + operator + " '" + formattedvalue + "'";
+  }
+
+  @Override
+  public String getDatePredicateCondition(String column, long value, String valueFormat, String operator) {
+    log.debug("Getting date predicate for Oracle");
+    String formattedvalue = Utils.toDateTimeFormat(Long.toString(value), valueFormat, DATE_FORMAT);
+    return Utils.getCoalesceColumnNames(column) + " " + operator + " '" + formattedvalue + "'";
```

Comment:
@lbendig, Can you elaborate on your comment? I borrowed these methods from `MySqlExtractor`. Let me know how you think `to_date` should be used, and I can refactor these methods accordingly.